Indicators/StochasticRelativeStrengthIndex.cs

```diff
@@ -0,0 +1,128 @@
+
+        /// <summary>
+        /// Gets the %K output
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> K { get; }
+
+        /// <summary>
+        /// Gets the %D output
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> D { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the StochasticRelativeStrengthIndex class
+        /// </summary>
+        /// <param name="rsiPeriod">The period of the relative strength index</param>
+        /// <param name="stochPeriod">The period of the stochastic indicator</param>
+        /// <param name="kSmoothingPeriod">The smoothing period of K output (aka %K)</param>
+        /// <param name="dSmoothingPeriod">The smoothing period of D output (aka %D)</param>
+        /// <param name="movingAverageType">The type of moving average to be used for k and d</param>
+        public StochasticRelativeStrengthIndex(int rsiPeriod, int stochPeriod, int kSmoothingPeriod, int dSmoothingPeriod, MovingAverageType movingAverageType = MovingAverageType.Simple)
+            : this($"StochRSI({rsiPeriod},{stochPeriod},{kSmoothingPeriod},{dSmoothingPeriod},{movingAverageType})", rsiPeriod, stochPeriod, kSmoothingPeriod, dSmoothingPeriod, movingAverageType)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the StochasticRelativeStrengthIndex class
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="rsiPeriod">The period of the relative strength index</param>
+        /// <param name="stochPeriod">The period of the stochastic indicator</param>
+        /// <param name="kSmoothingPeriod">The smoothing period of K output</param>
+        /// <param name="dSmoothingPeriod">The smoothing period of D output</param>
+        /// <param name="movingAverageType">The type of moving average to be used</param>
+        public StochasticRelativeStrengthIndex(string name, int rsiPeriod, int stochPeriod, int kSmoothingPeriod, int dSmoothingPeriod, MovingAverageType movingAverageType = MovingAverageType.Simple)
+            : base(name)
+        {
+            _rsi = new RelativeStrengthIndex(rsiPeriod);
+            _recentRSIValues = new RollingWindow<decimal>(stochPeriod);
+
+            K = movingAverageType.AsIndicator($"{name}_K_{movingAverageType}", kSmoothingPeriod);
+            D = movingAverageType.AsIndicator($"{name}_D_{movingAverageType}", dSmoothingPeriod);
+
+            WarmUpPeriod = stochPeriod;
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples >= WarmUpPeriod;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
```

Comment:
- Please move these above constructor
- Hm not sure that `IsReady/WarmUpPeriod` values are correct 🤔, if I understand correctly: first need to wait till RSI ready, later (meaning +) wait till the rolling windows of RSI is full/ready, and finally (another +?) the moving averages? Basically IsReady when K & D are ready, and warmupPeriod is the sum of the periods?

Reply:
You catch very well problems @Martin-Molinero 👍 

With StochasticRelativeStrengthIndex(14, 14, 3, 3) we are having the following data

```
Date,Close,k,d
02/22/2013 12:00:00 AM,151.89,,
02/25/2013 12:00:00 AM,149.0,,
02/26/2013 12:00:00 AM,150.02,,
02/27/2013 12:00:00 AM,151.91,,
02/28/2013 12:00:00 AM,151.61,,
03/01/2013 12:00:00 AM,152.11,,
03/04/2013 12:00:00 AM,152.92,,
03/05/2013 12:00:00 AM,154.29,,
03/06/2013 12:00:00 AM,154.5,,
03/07/2013 12:00:00 AM,154.78,,
03/08/2013 12:00:00 AM,155.44,,
03/11/2013 12:00:00 AM,156.03,,
03/12/2013 12:00:00 AM,155.68,,
03/13/2013 12:00:00 AM,155.9,,
03/14/2013 12:00:00 AM,156.73,,
03/15/2013 12:00:00 AM,155.83,,
03/18/2013 12:00:00 AM,154.97,,
03/19/2013 12:00:00 AM,154.61,,
03/20/2013 12:00:00 AM,155.69,,
03/21/2013 12:00:00 AM,154.36,,
03/22/2013 12:00:00 AM,155.6,,
03/25/2013 12:00:00 AM,154.95,,
03/26/2013 12:00:00 AM,156.19,,
03/27/2013 12:00:00 AM,156.19,,
03/28/2013 12:00:00 AM,156.67,,
04/01/2013 12:00:00 AM,156.05,,
04/02/2013 12:00:00 AM,156.82,,
04/03/2013 12:00:00 AM,155.23,,
04/04/2013 12:00:00 AM,155.86,,
04/05/2013 12:00:00 AM,155.16,7.319288119108858,
04/08/2013 12:00:00 AM,156.21,21.591511595932086,
04/09/2013 12:00:00 AM,156.75,35.411391422876534,21.44073037930583
04/10/2013 12:00:00 AM,158.67,68.74472475620986,41.91587592500616
```

32 incoming values are required to have both values for k and d.

14 + 14 + 3 + 3 = 34
14 + 14 + max(3, 3) = 31